test/addons-abi/test_buffer/test_buffer.cc

```diff
@@ -0,0 +1,141 @@
+#include <string.h>
+#include <string>
+#include <node_api_helpers.h>
+
+#define JS_ASSERT(env, assertion, message) \
+  if (!(assertion)) { \
+    napi_throw_error((env), (std::string("assertion (" #assertion ") failed: ") + message).c_str()); \
```

Comment:
I don't understand. It does get concatenated with the `std::string` that precedes it.